src/lex.cpp

```diff
@@ -1238,6 +1246,74 @@ const char8* lexer::parse_hex_digits_and_underscores(
       [](char8 character) -> bool { return is_hex_digit(character); }, input);
 }
 
+QLJS_WARNING_PUSH
+QLJS_WARNING_IGNORE_GCC("-Wuseless-cast")
+const char8* lexer::parse_unicode_escape(const char8* input) noexcept {
+  const char8* escape_sequence_begin = input;
+  auto get_escape_span = [escape_sequence_begin, &input]() {
+    return source_code_span(escape_sequence_begin, input);
+  };
+
+  const char8* code_point_hex_begin;
+  const char8* code_point_hex_end;
+  if (input[2] == u8'{') {
+    code_point_hex_begin = &input[3];
+    input += 3;  // Skip "\u{".
+    bool found_non_hex_digit = false;
+    while (*input != u8'}') {
+      if (*input == '\0' && this->is_eof(input)) {
+        this->error_reporter_->report(error_unclosed_identifier_escape_sequence{
```

Comment:
Almost-blocking: This error is confusing in this context. This error is about escape sequences in *identifiers*, but this code is used for *string literals*.

I think we should add an enum to error_unclosed_identifier_escape_sequence to indicate whether the token is a template literal, a string literal, or an identifier. I'm fine if my suggestion is done in a separate PR, as long as note is made of it here (e.g. with a TODO comment).